source/Mlos.Python/mlos/Optimizers/RegressionModels/LassoCrossValidatedRegressionModel.py

```diff
@@ -135,19 +141,28 @@ def fit(self, feature_values_pandas_frame, target_values_pandas_frame, iteration
         x_df = self.one_hot_encoder_adapter.project_dataframe(feature_values_pandas_frame, in_place=False)
         y = target_values_pandas_frame[self.target_dimension_names].to_numpy()
         design_matrix = self._transform_x(x_df)
+
+        # ensure num_cross_validations < num_samples; and reinstantiate LassoCV regressor
+        if design_matrix.shape[0] < self.model_config.num_cross_validations:
+            self.lasso_model_kwargs['cv'] = design_matrix.shape[0] - 1
+            self._regressor = LassoCV(**self.lasso_model_kwargs)
+
         self._regressor.fit(design_matrix, y)
         self._trained = True
         self.last_refit_iteration_number = iteration_number
 
         # retain inverse(x.T * x) to use for confidence intervals on predicted values
         condition_number = np.linalg.cond(design_matrix)
-        if condition_number > 10.0 ** 10:
+        self.logger.info(
+            f'LassoCV: design_matrix condition number: {condition_number}'
+        )
+        if condition_number > 10.0 ** 4:
             # add small noise to x to remove singularity,
             #  expect prediction confidence to be reduced (wider intervals) by doing this
             self.logger.info(
-                f"Adding noise to design matrix used for prediction confidence due to condition number {condition_number} > 10^10."
+                f"Adding noise to design matrix used for prediction confidence due to condition number {condition_number} > 10^4."

```

Comment:
10**4
```suggestion
                f"Adding noise to design matrix used for prediction confidence due to condition number {condition_number} > 10**4."

```

Reply:
It's clear what you mean... but my CDO strongly suggests that we should stick to the Python exponentiation operator :)